test/common/routes_exist.go

```diff
@@ -0,0 +1,220 @@
+package common
+
+import (
+	"fmt"
+	"testing"
+
+	goctx "context"
+
+	routev1 "github.com/openshift/api/route/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type expectedRoute struct {
+	Name            string
+	isTLS           bool
+	ServiceName     string
+	IsGeneratedName bool
+}
+
+var expectedRoutes = map[string][]expectedRoute{
+	"3scale": []expectedRoute{
+		expectedRoute{
+			Name:  "backend",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:            "zync-3scale-api-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "apicast-staging",
+		},
+		expectedRoute{
+			Name:            "zync-3scale-api-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "apicast-production",
+		},
+		expectedRoute{
+			Name:            "zync-3scale-master-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "system-master",
+		},
+		expectedRoute{
+			Name:            "zync-3scale-provider-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "system-developer",
+		},
+		expectedRoute{
+			Name:            "zync-3scale-provider-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "system-provider",
+		},
+	},
+
+	"amq-online": []expectedRoute{
+		expectedRoute{
+			Name:  "console",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "standard-authservice",
+			isTLS: true,
+		},
+	},
+
+	"codeready-workspaces": []expectedRoute{
+		expectedRoute{
+			Name:  "codeready",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "devfile-registry",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "plugin-registry",
+			isTLS: true,
+		},
+	},
+
+	"middleware-monitoring-operator": []expectedRoute{
+		expectedRoute{
+			Name:  "alertmanager-route",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "grafana-route",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "prometheus-route",
+			isTLS: true,
+		},
+	},
+
+	"rhsso": []expectedRoute{
+		expectedRoute{
+			Name:  "keycloak",
+			isTLS: false,
+		},
+		expectedRoute{
+			Name:  "keycloak-edge",
+			isTLS: true,
+		},
+	},
+
+	"solution-explorer": []expectedRoute{
+		expectedRoute{
+			Name:  "tutorial-web-app",
+			isTLS: true,
+		},
+	},
+
+	"ups": []expectedRoute{
+		expectedRoute{
+			Name:  "ups-unifiedpush-proxy",
+			isTLS: true,
+		},
+	},
+
+	"user-sso": []expectedRoute{
+		expectedRoute{
+			Name:  "keycloak",
+			isTLS: false,
+		},
+		expectedRoute{
+			Name:  "keycloak-edge",
+			isTLS: true,
+		},
+	},
+}
+
+const (
+	intlyNamespacePrefix = "redhat-rhmi-"
+)
```

Comment:
This constant looks to be already defined in `types.go` so just would be good to prevent duplicates :thinking: 
https://github.com/integr8ly/integreatly-operator/blob/8ff8ac219da0428dfbde872de8adbe104c39059d/test/common/types.go#L15